drivers/docker/driver.go

```diff
@@ -1109,12 +1109,22 @@ func (d *Driver) DestroyTask(taskID string, force bool) error {
 	if err != nil {
 		return fmt.Errorf("failed to inspect container state: %v", err)
```

Comment:
It feels that we should special case "container not found".  If a container was rm'ed out of band, we probably should run `cleanupImage` and run `d.tasks.Delete` anyway as task isn't running anymore.

Reply:
agreed. added test and updated: 
https://github.com/hashicorp/nomad/pull/5768/files#diff-87791a99732af8d5b49e0f329761a4cdR1615

Reply:
thanks, @notnoop